From 55ab3e708a37dded70aaca22921495a5dd54a479 Mon Sep 17 00:00:00 2001
From: Paul Seidler <pl.seidler@gmail.com>
Date: Sat, 16 Jul 2011 00:46:22 +0000
Subject: [PATCH] tests: readlink* ("",... fails with EINVAL on newer kernels
Upstream: yes, ported from gnulib, written by Jim Meyering

readlink and readlinkat have typically failed with ENOENT for
the invalid, empty file name,  "".  However, with the advent
of linux-2.6.39, they fail with EINVAL.
* tests/test-areadlink.h (test_areadlink): Also accept EINVAL
when operating on the empty file name.
* tests/test-readlink.h (test_readlink): Likewise.
---
 gettext-tools/gnulib-tests/test-readlink.h  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tests/test-readlink.h b/tests/test-readlink.h
index 1e482df..ae37b68 100644
--- a/tests/test-readlink.h
+++ b/tests/test-readlink.h
@@ -38,7 +38,7 @@ test_readlink (ssize_t (*func) (char const *, char *, size_t), bool print)
   ASSERT (errno == ENOENT);
   errno = 0;
   ASSERT (func ("", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOENT);
+  ASSERT (errno == ENOENT || errno == EINVAL);
   errno = 0;
   ASSERT (func (".", buf, sizeof buf) == -1);
   ASSERT (errno == EINVAL);
-- 
1.7.6

